src/ucp/rndv/proto_rndv.c

```diff
@@ -0,0 +1,325 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "proto_rndv.h"
+
+#include <ucp/proto/proto_common.inl>
+
+
+static ucp_md_map_t
+ucp_proto_rndv_ctrl_reg_md_map(const ucp_proto_rndv_ctrl_init_params_t *params)
+{
+    ucp_worker_h worker                      = params->super.super.worker;
+    const ucp_ep_config_key_t *ep_config_key = params->super.super.ep_config_key;
+    const uct_iface_attr_t *iface_attr;
+    const uct_md_attr_t *md_attr;
+    ucp_md_index_t md_index;
+    ucp_md_map_t reg_md_map;
+    ucp_lane_index_t lane;
+
+    /* md_map is all lanes which support get_zcopy on the given mem_type and
+     * require remote key
+     */
+    reg_md_map = 0;
+    for (lane = 0; lane < ep_config_key->num_lanes; ++lane) {
+        if (ep_config_key->lanes[lane].rsc_index == UCP_NULL_RESOURCE) {
+            continue;
```

Comment:
Why? i'd prefer to not add any wireup-related logic to protocols. 
The protocols should only look at wireup outcome and not examine internal wireup logic

Reply:
I think we want to check that NULL_RESOURCE can be only for CM lane

Reply:
but then if we change wireup logic so that a lane could be NULL_RESOURCE for some other reason, it means that changes in internal wireup logic (which do not affect wireup final result - ep lanes), will have to change protocols code?